configapi/api_slice_mgmt.go

```diff
@@ -198,15 +200,28 @@ func NetworkSlicePostHandler(c *gin.Context, msgOp int) bool {
 	logger.ConfigLog.Infof("url: %v ", req.URL)
 	procReq := req.Body.(configmodels.Slice)
 
+	for _, gnb := range procReq.SiteInfo.GNodeBs {
+		if !isValidName(gnb.Name) {
+			err := fmt.Errorf("invalid gNB name `%s` in Network Slice %s. Name needs to match the following regular expression: %s", gnb.Name, sliceName, NAME_PATTERN)
+			logger.ConfigLog.Errorln(err.Error())
+			return err
+		}
+		castedGnbTac := fmt.Sprint(gnb.Tac)
+		if !isValidGnbTac(castedGnbTac) {
```

Comment:
Alternatively, would not it be easier/faster to avoid the conversion (and avoid calling `isValidGnbTac`) and just check the following, which is basically what `isValidName` is doing "under the hood"
```suggestion
		if gnb.Tac < 1 && gnb.Tac > 16777215 {
```

Reply:
I would change `Tac` in `PutGnbRequest` to `int32`, howeverr this is a breaking change which changes the behaviour of the API. If you are ok with that, I'll proceed.

Reply:
I think the API is mainly used by `Aether ROC/simapp` and `Canonical's NMS`, we need to validate that nothing breaks with these two because of the API change

Reply:
@gab-arrobo regarding Canonical NMS, we already have the PR to address this. Regarding `simapp`, it doesn't seem to use the `/api/config/v1/inventory/gnb` APIs, so this PR does not affect its behavior; it creates a NS but there the value for the TAC is already `1` (int) so the validation is positive.